.eslintrc.js

```diff
@@ -22,78 +56,393 @@ module.exports = {
     jsdom: true,
     L: true,
   },
+  overrides: [
+    {
+      files: ['**/*.test.js', 'src/test/utils.js'],
+      rules: {
+        'max-nested-callbacks': 'off',
+        'no-undef': 'off',
+        'require-await': 'off',
+        'promise/catch-or-return': 'off', // Shall we prefer await?
+        'promise/always-return': 'off',
+        'promise/no-callback-in-promise': 'off',
+        'no-magic-numbers': 'off',
+        'no-constructor-return': 'off',
+        'testing-library/prefer-find-by': 'off', // Needs some manual work
+        'promise/avoid-new': 'off',
+        'max-len': 'off',
+      },
+    },
+    {
+      files: ['internals/**/*'],
+      rules: {
+        'no-console': 'warn',
+      },
+    },
+    {
+      files: ['cypress/**/*'],
+      rules: {
+        'promise/always-return': 'off',
+        'promise/catch-or-return': 'off',
+      },
+    },
+  ],
   rules: {
-    camelcase: 'off',
-    'arrow-parens': ['error', 'as-needed', { requireForBlockBody: false }],
+    // proposed rules
+    'new-cap': 'off',
+    'func-names': 'off',
+    'unicorn/prefer-set-has': 'off',
+    'react/boolean-prop-naming': 'off',
+    'require-atomic-updates': 'off',
+    'no-negated-condition': 'off',
+    'no-import-assign': 'off',
+    'default-param-last': 'off',
+    'sonarjs/no-collapsible-if': 'off',
+    'id-length': 'off',
+    'unicorn/no-abusive-eslint-disable': 'off',
+    'unicorn/prefer-includes': 'off', // Has autofix but breaks test
+    'unicorn/prefer-text-content': 'off', // Has autofix but breaks snapshots test
+    'prefer-named-capture-group': 'off',
+    'no-warning-comments': 'off',
+    'unicorn/prefer-add-event-listener': 'off',
+    'no-magic-numbers': 'off', // Except for tests
+    'init-declarations': 'off',
+    'sonarjs/no-duplicate-string': 'off',
+    'require-unicode-regexp': 'off',
+    'line-comment-position': 'off',
+    'react/require-optimization': 'off',
+    'sonarjs/no-small-switch': 'off',
+    'no-implicit-coercion': 'off',
+    'react/display-name': 'off',
+    'unicorn/prevent-abbreviations': 'off',
+    'unicorn/catch-error-name': 'off',
+    'react/jsx-key': 'off',
+    'sonarjs/no-identical-functions': 'off',
+    'no-promise-executor-return': 'off',
+    'prefer-regex-literals': 'off',
+    'unicorn/no-zero-fractions': 'off', // Auto-fix available
+    'max-len': 'off', // 'max-len': ['warn', { code: 120, tabWidth: 2, ignoreUrls: true, comments: 140 }],
```

Comment:
`max-len` Should be set to 120 characters. What is the impact of enabling this rule?

Reply:
Quite some (manual) changes needs to be done which I kept out of scope for this pull request because of time.